solidity/contracts/ECDSARewardsDistributor.sol

```diff
@@ -78,17 +83,21 @@ contract ECDSARewardsDistributor is Ownable {
         require(!isClaimed(merkleRoot, index), "Reward already claimed");
 
         // Verify the merkle proof.
-        bytes32 node = keccak256(abi.encodePacked(index, account, amount));
+        bytes32 node = keccak256(abi.encodePacked(index, operator, amount));
         require(
             MerkleProof.verify(merkleProof, merkleRoot, node),
             "Invalid proof"
         );
 
         // Mark it claimed and send the token.
         _setClaimed(merkleRoot, index);
-        require(IERC20(token).transfer(account, amount), "Transfer failed");
 
-        emit RewardsClaimed(merkleRoot, index, account, amount);
+        address beneficiary = tokenStaking.beneficiaryOf(operator);
+        require(beneficiary != address(0), "Beneficiary address not set");
```

Comment:
I'm not sure if we actually need this check:
- is it possible that we have an operator without beneficiary?
- I think `transfer` already [checks](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L210) for `address(0)`

Reply:
I initially had the same thoughts. On the other hand, it makes sense to treat this contract independently and this contract does not have to be aware of the rules of `TokenStaking` has. 

For the second point, that's true but I think that printing a more targeted revert makes sense.

Reply:
I think that a check for zero address should be there. As this is already covered by a library we use underneth it's redundant to have a separate require. The message returned by the OZ's library is very fine: `ERC20: transfer to the zero address`. I'll remove the require statement and update tests.

Reply:
I'm okay with both. I know it wouldn't hurt us to check and write a targeted message and I take that the gas usage should be marginal.

Reply:
`ERC20: transfer to the zero address` message is detailed enough. I haven't checked what it is and assumed it's more generic.

Reply:
fea7a4800752ddbda01ddf7423aa73431e5e502d